src/python/pants/base/exception_sink.py

```diff
@@ -371,11 +371,15 @@ def trapped_signals(cls, new_signal_handler):
 
     NB: This method calls signal.signal(), which will crash if not called from the main thread!
     """
+    # NB: We set `previous_signal_handler` to None at the start of the method in case resetting the
+    # signal handler within the `try` raises an error for some reason.
+    previous_signal_handler = None
```

Comment:
Thanks for writing this down. So, if it does raise, then `previous_signal_handler` is None and the finally block no-ops. As such... isn't this exactly the same as?:
```python
previous_signal_handler = cls.reset_signal_handler(new_signal_handler)
try:
  yield
finally:
  cls.reset_signal_handler(previous_signal_handler)
```
```

Reply:
This logic was more difficult for me to follow than usual for some reason, but I believe you are correct! I will make this change!